From 1e9651d6bea39ce7391fe6fc34a93366b41c6856 Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kai.uwe.broulik@basyskom.com>
Date: Tue, 24 Sep 2024 11:58:06 +0200
Subject: [PATCH] lookandfeel: Explicitly set theme to Breeze in defaults

Currently, we only set "library" to "org.kde.breeze", which means
for "theme" the fallback mechanism is used. Depending on how
plasma-workspace was compiled, this is either Breeze or
kwin4_decoration_qml_plastik.

When plasma-workspace was built without Breeze, we end up
with library=org.kde.breeze, theme=kwin4_decoration_qml_plastik,
which is not a valid combination. While this will ultimately result
in Breeze decoration being used as intended, it breaks the
decoration KCM highlight since it tries to select a combination
of library/theme that doesn't exist. KWin might also get confused,
its support information also prints this bogus configuration.

While this doesn't fix the underlying issue of defaults handling
in lnf (see also !3345) and might also indicate a packaging issue
of plasma-workspace, it is imho still advisable to be explicit.

Signed-off-by: Sebastian Stark <sebastian.stark@advantest.com>
(cherry picked from commit c520c44132c4033c8bfd45c7b4aaf8e47e8612ed)
(cherry picked from commit 2c651a920a446037967ac2dfdc21fc8da6ae2939)
---
 lookandfeel/org.kde.breeze/contents/defaults         | 1 +
 lookandfeel/org.kde.breezedark/contents/defaults     | 1 +
 lookandfeel/org.kde.breezetwilight/contents/defaults | 1 +
 3 files changed, 3 insertions(+)

diff --git a/lookandfeel/org.kde.breeze/contents/defaults b/lookandfeel/org.kde.breeze/contents/defaults
index d8df939b37c..7a2c811bbaa 100644
--- a/lookandfeel/org.kde.breeze/contents/defaults
+++ b/lookandfeel/org.kde.breeze/contents/defaults
@@ -24,6 +24,7 @@ LayoutName=org.kde.breeze.desktop
 
 [kwinrc][org.kde.kdecoration2]
 library=org.kde.breeze
+theme=Breeze
 
 [KSplash]
 Theme=org.kde.Breeze
diff --git a/lookandfeel/org.kde.breezedark/contents/defaults b/lookandfeel/org.kde.breezedark/contents/defaults
index 3db126ff2a6..eee919a436b 100644
--- a/lookandfeel/org.kde.breezedark/contents/defaults
+++ b/lookandfeel/org.kde.breezedark/contents/defaults
@@ -24,6 +24,7 @@ LayoutName=org.kde.breeze.desktop
 
 [kwinrc][org.kde.kdecoration2]
 library=org.kde.breeze
+theme=Breeze
 
 [KSplash]
 Theme=org.kde.Breeze
diff --git a/lookandfeel/org.kde.breezetwilight/contents/defaults b/lookandfeel/org.kde.breezetwilight/contents/defaults
index 02bc44546df..b35cef1562c 100644
--- a/lookandfeel/org.kde.breezetwilight/contents/defaults
+++ b/lookandfeel/org.kde.breezetwilight/contents/defaults
@@ -24,6 +24,7 @@ LayoutName=org.kde.breeze.desktop
 
 [kwinrc][org.kde.kdecoration2]
 library=org.kde.breeze
+theme=Breeze
 
 [KSplash]
 Theme=org.kde.Breeze
-- 
GitLab

